set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java

```diff
@@ -1,557 +1,600 @@
+import eu.europeana.set.definitions.model.vocabulary.VisibilityTypes;
+import eu.europeana.set.definitions.model.vocabulary.WebUserSetFields;
+import eu.europeana.set.definitions.model.vocabulary.fields.WebUserSetModelFields;
+import eu.europeana.set.mongo.model.internal.PersistentUserSet;
+import eu.europeana.set.search.exception.SearchApiClientException;
+import eu.europeana.set.search.service.SearchApiClient;
+import eu.europeana.set.search.service.SearchApiResponse;
+import eu.europeana.set.search.service.impl.SearchApiClientImpl;
+import eu.europeana.set.web.exception.request.RequestBodyValidationException;
+import eu.europeana.set.web.exception.response.UserSetNotFoundException;
+import eu.europeana.set.web.model.WebUserSetImpl;
+import eu.europeana.set.web.service.UserSetService;
+import ioinformarics.oss.jackson.module.jsonld.JsonldModule;
+
+public class UserSetServiceImpl extends BaseUserSetServiceImpl implements UserSetService {
+
+    @Resource
+    I18nService i18nService;
+
+    UserSetUtils userSetUtils = new UserSetUtils();
+
+    private SearchApiClient setApiService = new SearchApiClientImpl();
+
+    public UserSetUtils getUserSetUtils() {
+	return userSetUtils;
+    }
+
+    public SearchApiClient getSearchApiClient() {
+	return setApiService;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see eu.europeana.UserSet.web.service.UserSetService#storeUserSet(eu.
+     * europeana.UserSet.definitions.model.UserSet)
+     */
+    @Override
+    public UserSet storeUserSet(UserSet newUserSet) {
+
+	UserSet extUserSet = getUserSetUtils().analysePagination(newUserSet);
+
+	// store in mongo database
+	UserSet res = getMongoPersistence().store(extUserSet);
+	return res;
+    }
+
+    @Override
+    public void updateUserSetPagination(UserSet newUserSet) {
+	getUserSetUtils().updatePagination(newUserSet);
+    }
+
+    @Override
+    public UserSet getUserSetById(String userSetId) throws UserSetNotFoundException {
+	return getUserSetById(userSetId, true);
+    }
+
+    @Override
+    public UserSet getUserSetById(String userSetId, boolean checkDisabled) throws UserSetNotFoundException {
+	UserSet res = getMongoPersistence().getByIdentifier(userSetId);
+	if (res == null) {
+	    throw new UserSetNotFoundException(I18nConstants.USERSET_NOT_FOUND, I18nConstants.USERSET_NOT_FOUND,
+		    new String[] { userSetId });
+	} else if (checkDisabled && res.isDisabled()) {
+	    throw new UserSetNotFoundException(I18nConstants.USER_SET_NOT_AVAILABLE,
+		    I18nConstants.USER_SET_NOT_AVAILABLE, new String[] { userSetId }, HttpStatus.GONE);
+	}
+	return res;
+    }
+
+    /**
+     * This method checks if a user set with provided type and user already exists in database
+     * @param type
+     * @param creator
+     * @return true if a user set object found in database
+     * @throws UserSetNotFoundException
+     */
+    public boolean getBookmarksFolder(String type, Agent creator) throws UserSetNotFoundException {
```

Comment:
please return the user set object or null and remove the type parameter

Reply:
User set object is returned now and type parameter removed